awx/main/dispatch/periodic.py

```diff
@@ -38,10 +39,21 @@ def run():
                         # connection
                         conn.close_if_unusable_or_obsolete()
                     set_guid(generate_guid())
-                    self.run_pending()
+
+                    # same logic as run_pending but split up to give spacing between jobs
+                    # when more than 1 job is available to run now, put spacing between them
+                    # this is to spread out pressure on dispatcher worker pool,
+                    # avoiding need for excess reserve workers
+                    runnable_jobs = sorted(job for job in self.jobs if job.should_run)
+                    for job in runnable_jobs[:-1]:
+                        self._run_job(job)
+                        time.sleep(0.3)
+                    if runnable_jobs:
+                        self._run_job(runnable_jobs[-1])
+
                 except Exception:
                     logger.exception('encountered an error while scheduling periodic tasks')
-                time.sleep(idle_seconds)
+                time.sleep(idle_seconds - (time.time() - loop_start))
```

Comment:
The pre-existing code here seems extraordinarily lazy... as there is a documented way to do this

https://schedule.readthedocs.io/en/stable/examples.html#time-until-the-next-execution

Here, that would be

```python
time.sleep(self.idle_seconds())
```

Why on Earth we would not start out using this, I do not know, and I am a bit dumbfounded about how the library is completely circumvented with an obviously less-good alternative. This demands some remedy my changes here, because I'm adding additional sleeps and any weird skews this previously caused would worsen if this line isn't changed.